pkg/certificate/providers/vault/tools.go

```diff
@@ -15,9 +15,18 @@ func getIssueURL(role string) string {
 	return fmt.Sprintf("pki/issue/%+v", role)
 }
 
-func getIssuanceData(cn certificate.CommonName, validityPeriod time.Duration) map[string]interface{} {
-	return map[string]interface{}{
-		commonNameField: cn.String(),
-		ttlField:        getDurationInMinutes(validityPeriod),
+func getIssuanceData(options certificate.IssueOptions) map[string]interface{} {
+	issuanceData := map[string]interface{}{
+		commonNameField: options.CommonName().String(),
+		ttlField:        getDurationInMinutes(options.ValidityDuration),
 	}
+
+	if options.SpiffeEnabled {
+		log.Trace().Str("cn", options.CommonName().String()).Msg("Generating SPIFFE enabled Certificate")
+
+		// Comma delimited list; For SPIFFE compatibility it should only be one.
+		issuanceData[uriSans] = options.SpiffeURL().String()
```

Comment:
would it be possible to change the `CommonName()` method to generate the SPIFFE URL if SPIFFE is enabled.

And to always populate issuanceData with the uriSans = options.CommonName()?

That way the providers can stay "dumb" and not need to know about spiffe?

Reply:
I didn't do this since we are still using common name to do validation, which is required to be able to do an upgrade in backwards compat way, i.e. issue all certs with SPIFFE URI SANs (via a MRC rotation) then turn on validation using that field (#5025 and #5023)

Another aspect of this is, the Ingresses we support only currently do validation via the DNS SAN (nginx and contour) which will be in the format of the common name.

One thing I considered was to do something like `if options.SpiffeId() != "" then urisans = options.spiffeId` which would remove the extra `SpiffeEnabled`. Maybe making it `options.UriSan() != ""` could keep them "dumb"

Reply:
gotcha ya that makes sense. Up to you on your last point, i'm relatively indifferent there. Although I do like the idea of abstracting Spiffe away for the UriSan method

Reply:
I do too.  I'll make this update

Reply:
updated